src/metabase/pulse.clj

```diff
@@ -209,26 +224,44 @@
         (str "…"))
     mrkdwn))
 
+(defn- text->markdown-block
+  [text]
+  (let [mrkdwn (markdown/process-markdown text :slack)]
+    (when (not (str/blank? mrkdwn))
+      {:blocks [{:type "section"
+                 :text {:type "mrkdwn"
+                        :text (truncate-mrkdwn mrkdwn block-text-length-limit)}}]})))
+
+(defn- part->attachment-data
+  [part channel-id]
+  (let [{ttype                                  :type
+         {card-id :id card-name :name :as card} :card
+         dashcard                               :dashcard
+         result                                 :result}  part]
```

Comment:
I still think these are only relevant to the `type=card` case so they shouldn't be destructured at the top. As a general rule for readability, prefer to bind names closer to where they're used, and avoid binding names that are not used

Reply:
yep ba6cefff13